### ISSUE TITLE
.eslintignore **/*.js also ignores **/*.jsx 

### ISSUE THREAD

**Issue by sbason** [eslint/eslint#9660]
* **ESLint Version:** 4.12.0
* **Node Version:** 6
* **npm Version:** 5

**What parser (default, Babel-ESLint, etc.) are you using?** N/A

When ignoring folder/**/*.js, JSX files in the ignored folders are also not linted when `eslint .` is run.

```js

node_modules
coverage
views/**/*.js
```

**Comment by sbason** [comment]
1. ESLint Version: 4.12.0

2. I ran `$ eslint .` when there was a linting error I had introduced in folder views/xyz.jsx
My config is set in my package.json as follows:
``` 
"eslintConfig": {
    "extends": "iplayer-jsx/inferno"
  },
```
3. I got no output 
```
> eslint .

```
4. When I expected a linting error.

**Comment by sbason** [comment]
As above my .eslintignore is 
```js

node_modules
coverage
views/**/*.js
```

**Comment by platinumazure** [comment]
When running ESLint against a directory (`.`), we convert that internally into a glob pattern based on the extensions you have specified using the [`--ext` flag](https://eslint.org/docs/user-guide/command-line-interface#--ext). As you haven't specified the `--ext` flag, ESLint runs on JS files only by default.

Possible solutions:

1. Specify the extensions: `eslint . --ext js,jsx` or `eslint . --ext js --ext jsx`
1. Use glob patterns instead: `eslint "./**/*.{js,jsx}"`

One way you can check that this is indeed the issue is to temporarily remove/comment out the JS ignore pattern from your .eslintignore file. The JSX files should still not be linted, simply because ESLint should lint JS files by default. So in reality, this doesn't seem to be an eslintignore problem but rather a problem with telling ESLint what files it should operate on.

**Comment by sbason** [comment]
Perfect. Thanks. Solved my issue!
